src/analysis/normed_space/linear_isometry.lean

```diff
@@ -415,4 +415,27 @@ variables {R}
 
 @[simp] lemma symm_neg : (neg R : E ≃ₗᵢ[R] E).symm = neg R := rfl
 
+variables (R E E₂ E₃)
+
+/-- The natural equivalence `(E × E₂) × E₃ ≃ E × (E₂ × E₃)` is a linear isometry. -/
+noncomputable def prod_assoc [module R E₂] [module R E₃] : (E × E₂) × E₃ ≃ₗᵢ[R] E × E₂ × E₃ :=
+{ to_fun    := equiv.prod_assoc E E₂ E₃,
+  inv_fun   := (equiv.prod_assoc E E₂ E₃).symm,
+  map_add'  := by simp,
+  map_smul' := by simp,
+  norm_map' :=
+    begin
+      rintros ⟨⟨e, f⟩, g⟩,
+      simp only [linear_equiv.coe_mk, equiv.prod_assoc_apply, prod.semi_norm_def, max_assoc],
+    end,
+  .. equiv.prod_assoc E E₂ E₃, }
+
+@[simp] lemma coe_prod_assoc [module R E₂] [module R E₃] :
+  (prod_assoc R E E₂ E₃ : (E × E₂) × E₃ → E × E₂ × E₃) = equiv.prod_assoc E E₂ E₃ :=
+rfl
```

Comment:
If you set up the simps lemmas for `linear_isometry_equiv` this lemma will be generated automatically with `@[simps]`.
To do this, mimic
https://github.com/leanprover-community/mathlib/blob/16af388f7ea22fcdf7260f6cb340562aae0654aa/src/topology/local_homeomorph.lean#L83-L91

You can even get this exact name if you use the name `coe as_prefix` instead of `apply`.
(if you want (only) this exact lemma you need `@[simps coe as_fn]`

Reply:
I wrestled with this for more than 15 minutes but I couldn't get this to work :-/

Reply:
Oh wait, I missed that you're stating an equality between `equiv`s, not functions. This is possible with `simps`, but not super easy, so let's leave it for now.

bors d+

Reply:
OK thanks!